src/victory-util/collection.js

```diff
@@ -95,5 +95,50 @@ export default {
     return itemSets.every((comparisonSet) => {
       return isEqual(comparisonSet[0], comparisonSet[1]);
     });
+  },
+
+  /*
+  Custom equality checking for props in shouldComponentUpdate.
+  `areVictoryPropsEqual` differs from lodash `isEqual` in the following ways:
+    - Functions are marked as equal (this was the main impetus for writing a new function)
+    - Does not handle symbols or maps
+    - Returns false when checking the equality of things like `1` vs. `Object(1)`
+    - Does not handle circular references in objects and arrays
+  */
+  areVictoryPropsEqual(a, b) {
+    return this.checkEquality(a, b);
+  },
+
+  basicEqualityCheck(a, b) {
+    if (a === b) { return true; }
+    if (typeof a !== typeof b) { return false; }
+    // isEqual does not support equality checking on functions
+    // return true if a and b are both functions
+    if (typeof a === "function") { return true; }
+    if (typeof a === "object" && keys(a).length !== keys(b).length) { return false; }
+    if (typeof a !== "object" || keys(a).length === 0) { return isEqual(a, b); }
+    return undefined;
```

Comment:
We should document this internal function to say that it's tri-state and spell out that you cannot just use it straight up without comparing the type of the return value.